src/System.Private.CoreLib/shared/System/Text/Encoding.cs

```diff
@@ -817,8 +810,7 @@ public byte[] GetBytes(string s, int index, int count)
                 byteCount = result;
 
             // Copy the data, don't overrun our array!
-            for (index = 0; index < byteCount; index++)
-                bytes[index] = arrByte[index];
+            new ReadOnlySpan(arrByte).CopyTo(new Span<byte>(bytes, byteCount));
```

Comment:
```suggestion
            new ReadOnlySpan(arrByte, 0, byteCount).CopyTo(new Span<byte>(bytes, byteCount));
```

It's possible that `byteCount`'s value will be modified by the `GetBytes` method (see lines 804 - 810). Make sure to slice the source span appropriately before copying to the destination.